src/components/slide/selection-frame.tsx

```diff
@@ -221,48 +205,83 @@ export const SelectionFrame: React.FC<Props> = ({ children, treeId }) => {
 
   const isHovered = hoveredElementId === children.props.id;
   const isSelected = editableElementId === children.props.id;
-  const isSelectedAndMarkdown = isSelected && isMdElement(selectedElement);
+
+  const handleMouseDown = useCallback(
+    (event: MouseEvent<HTMLDivElement>) => {
+      if (
+        (event.target as HTMLElement).classList.contains('moveable-control')
+      ) {
+        return;
+      }
+      event.stopPropagation();
+      dispatch(deckSlice.actions.editableElementSelected(children.props.id));
+
+      if (event.detail === 2) {
+        setDoubleClickedElement(true);
+      }
+    },
+    [children.props.id, dispatch]
+  );
+
+  useEffect(() => {
+    if (doubleClickedElement && selectedElement?.id !== children.props.id) {
+      setDoubleClickedElement(false);
+    }
+  }, [children.props.id, doubleClickedElement, selectedElement?.id]);
+
+  const isEditingMarkdown =
+    doubleClickedElement && isMdElement(selectedElement);
+
+  const editingFrameMetrics = children.props.componentProps?.isFreeMovement
+    ? {
+        left: children?.props?.componentProps?.positionX || 0,
+        top: children?.props?.componentProps?.positionY || 0,
+        width: children?.props?.width,
+        height: children?.props?.height
+      }
+    : {};
 
   return (
     <>
-      <Wrapper
-        isHovered={isHovered}
-        isSelected={isSelected}
+      <div
+        className={clsx({
+          'wrapper selected': isSelected,
+          'wrapper hovered': isHovered
```

Comment:
Just to double-check: is it ok that it doesn't have the `wrapper` class when it is neither selected nor hovered?

Reply:
Correct. I can add it as a default and drop the conditional wrapper but it does nothing when it's not selected or hovered.